tests/test_operation.py

```diff
@@ -975,6 +975,88 @@ def test_prune(self, tensor_observable, expected, statistics):
         assert O_pruned.wires == expected.wires
         assert O_pruned.return_type == O_expected.return_type
 
+
+equal_obs = [
+    (qml.PauliZ(0), qml.PauliZ(0), True),
+    (qml.PauliZ(0) @ qml.PauliX(1), qml.PauliZ(0) @ qml.PauliX(1) @ qml.Identity(2), True),
+    (qml.PauliZ("b"), qml.PauliZ("b") @ qml.Identity(1.3), True),
+    (qml.PauliZ(0) @ qml.Identity(1), qml.PauliZ(0), True),
+    (qml.PauliZ(0), qml.PauliZ(1) @ qml.Identity(0), False),
+    (qml.Hermitian(np.array([[0, 1], [1, 0]]), 0),
+     qml.Identity(1) @ qml.Hermitian(np.array([[0, 1], [1, 0]]), 0), True),
+    (qml.PauliZ("a") @ qml.PauliX(1), qml.PauliX(1) @ qml.PauliZ("a"), True)
+]
```

Comment:
Since `Observable.compare()` will accept Hamiltonians, make sure to include a Hamiltonian here!